chain/chunks/src/logic.rs

```diff
@@ -245,18 +245,14 @@ pub fn persist_chunk(
     shard_chunk: Option<ShardChunk>,
     store: &mut ChainStore,
 ) -> Result<(), Error> {
-    let need_persist_partial_chunk = store.get_partial_chunk(&partial_chunk.chunk_hash()).is_err();
-    let need_persist_shard_chunk = shard_chunk.is_some()
-        && store.get_chunk(&shard_chunk.as_ref().unwrap().chunk_hash()).is_err();
     let mut update = store.store_update();
-    if need_persist_partial_chunk {
+    if update.get_partial_chunk(&partial_chunk.chunk_hash()).is_err() {
```

Comment:
Instead of calling `update.get_*`, could we call `store.get_*`? The idea is to move away from the `ChainStoreUpdate` implementing getter functionality

Reply:
When we hold `update`, we have a `mut` reference to `store` so we cannot call the getters on it.